test_spack.py

```diff
@@ -0,0 +1,432 @@
+    machines = all_machines
+
+    def test_install_master_gpu(self):
+        # So our quick start tutorial works: https://c2sm.github.io/spack-c2sm/QuickStart.html
+        run('spack installcosmo cosmo@master%pgi cosmo_target=gpu +cppdycore')
+
+    def test_install_master_cpu(self):
+        # So our quick start tutorial works: https://c2sm.github.io/spack-c2sm/QuickStart.html
+        run('spack installcosmo cosmo@master%pgi cosmo_target=cpu ~cppdycore')
+
+    # def test_install_test(self):
+    #     # TODO: Decide if we want to integrate this test or not. It has been used lately here: From https://github.com/C2SM/spack-c2sm/pull/289
+    #     run('spack installcosmo --test=root cosmo@master%pgi')
+
+    # def test_install_test_claw(self):
+    #     # TODO: Decide if we want to integrate this test or not. It has been used lately here: From https://github.com/C2SM/spack-c2sm/pull/289
+    #     run('spack installcosmo --test=root cosmo@master%pgi +claw')
+
+    def test_devbuild_cpu(self):
+        # So our quick start tutorial works: https://c2sm.github.io/spack-c2sm/QuickStart.html
+        run('git clone git@github.com:MeteoSwiss-APN/cosmo.git')
+        try:
+            run('spack devbuildcosmo cosmo@dev-build%pgi cosmo_target=cpu ~cppdycore', cwd='cosmo')
+        finally:
+            run('rm -rf cosmo')
+
+    def test_devbuild_gpu(self):
+        # So our quick start tutorial works: https://c2sm.github.io/spack-c2sm/QuickStart.html
+        run('git clone git@github.com:MeteoSwiss-APN/cosmo.git')
+        try:
+            run('spack devbuildcosmo cosmo@dev-build%pgi cosmo_target=gpu +cppdycore', cwd='cosmo')
+        finally:
+            run('rm -rf cosmo')
+
+
+class CosmoDycoreTest(unittest.TestCase):
+    package_name = 'cosmo-dycore'
+    depends_on = {'gridtools', 'serialbox', 'cuda'}
+    machines = all_machines
+
+
+class CosmoEccodesDefinitionsTest(unittest.TestCase):
+    package_name = 'cosmo-eccodes-definitions'
+    depends_on = {'eccodes'}
+    machines = all_machines
+
+
+class CosmoGribApiTest(unittest.TestCase):
+    package_name = 'cosmo-grib-api'
+    depends_on = {}
+    machines = all_machines
+
+
+class CosmoGribApiDefinitionsTest(unittest.TestCase):
+    package_name = 'cosmo-grib-api-definitions'
+    depends_on = {'cosmo-grib-api'}
+    machines = all_machines
+
+
+class CudaTest(unittest.TestCase):
+    package_name = 'cuda'
+    depends_on = {}
+    machines = all_machines
+
+
+class DawnTest(unittest.TestCase):
+    package_name = 'dawn'
+    depends_on = {}
+    machines = all_machines
+
+
+class Dawn4PyTest(unittest.TestCase):
+    package_name = 'dawn4py'
+    depends_on = {}
+    machines = all_machines
+
+
+class DuskTest(unittest.TestCase):
+    package_name = 'dusk'
+    depends_on = {'dawn4py'}
+    machines = all_machines
+
+
+class DyiconBenchmarksTest(unittest.TestCase):
```

Comment:
this one I think we can decomission, there is no jenkins plan for it. So we should not support it here